doc/guides/modules.rst

```diff
@@ -10,14 +10,380 @@ system those are called *modules*. These modules must be integrated with the
+
+
+Maintaining module code base
+============================
+
+Updates in the Zephyr main tree, for example, in public Zephyr APIs,
+may require patching a module code base. The responsibility for keeping
+the module code base up to date is shared between the **contributor** of
+such updates in Zephyr and the module **owner**. In particular:
+
+* the contributor of the original changes in Zephyr is required to submit
+  the corresponding patches that are required in module repositories, to
+  ensure that Zephyr CI on the pull request with the original changes, as
+  well as the module integration testing are successful.
+
+* the module owner has the overall responsibility for synchronizing
+  and testing the module code base with the Zephyr main tree.
+  This includes occasional advanced testing of the module code base
+  in addition to the testing performed by Zephyr's CI.
+  The module owner is required to fix issues in the module code base that
+  have not been caught by Zephyr pull request CI runs.
+
+  .. note::
+
+    The above polices apply on all Zephyr modules. However, modules which
+    Zephyr depends on, for example vendor HALs, are more crucial to be kept
+    up-to-date with Zephyr main tree.
+
+
+
+Contributing changes to modules
+===============================
+
+Submitting and merging patches directly to a module's code base, that is,
+before they have been merged in the corresponding external project
+repository, should be limited to:
+
+* patches required due to updates in the Zephyr main tree
+* urgent patches that should not wait to be merged in the external project
+  first, such as fixes to security vulnerabilities.
+
+Non trivial changes to module code base, including changes in the module
+design or functionality should be discouraged, if the module has an upstream
+project repository. In that case, such changes shall be submitted to the
+upstream project, directly.
+
+:ref:`Submitting changes to modules <submitting_new_modules>` describes in
+detail the process of contributing changes to module repositories.
+
+Contribution guidelines
+-----------------------
+
+Contributing to Zephyr modules shall follow the generic project
+:ref:`Contribution guidelines <contribute_guidelines>`.
+
+**Pull Requests:** may be merged with minimum of 2 approvals, including
+an approval by the PR assignee. In addition to this, pull requests in module
+repositories may only be merged, if the introduced changes are verified
+with Zephyr CI tools, as described in more detail in other sections on
+this page.
+
+**Issue Reporting:** intentionally disabled in module repositories, in
+favor of a centralized policy for issue reporting. Tickets concerning, for
+example, bugs or enhancements in modules shall be opened in the main
+Zephyr repository. Issues should be appropriately labeled using Github
+labels corresponding to each module, when applicable.
+
+  .. note::
+
+     Bug reports in modules, mirroring upstream project bug reports, are
```

Comment:
What do you mean here? I did not understand this

Reply:
Thanks
Changed to
```

     It is allowed to file bug reports for zephyr modules to track
     the corresponding upstream project bugs in Zephyr.
```